iidm/iidm-api/src/main/java/com/powsybl/iidm/network/Boundary.java

```diff
@@ -6,16 +6,63 @@
  */
 package com.powsybl.iidm.network;
 
+import com.powsybl.commons.PowsyblException;
+
 /**
  * @author Miora Ralambotiana <miora.ralambotiana at rte-france.com>
  */
 public interface Boundary {
 
+    interface BoundaryTerminal extends Terminal {
+
+        @Override
+        default Terminal.NodeBreakerView getNodeBreakerView() {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default Terminal.BusBreakerView getBusBreakerView() {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default Terminal.BusView getBusView() {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default BoundaryTerminal setP(double p) {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default BoundaryTerminal setQ(double q) {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default boolean connect() {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default boolean disconnect() {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+
+        @Override
+        default void traverse(VoltageLevel.TopologyTraverser traverser) {
+            throw new PowsyblException("Not supported for boundary terminals");
+        }
+    }
+
     double getV();
 
     double getAngle();
 
     double getP();
```

Comment:
You prefer to leave these functions even if they are redondant with the ones accessible from the terminal? Then, I imagine that it is too complex to have a bus view from the boundary terminal, that is why you the access to V and angle remains from boundary object?